scripts/what_changed.py

```diff
@@ -0,0 +1,345 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# A script to generate a list of boards that have changed or added and create an
+# arguemnts file for sanitycheck to allow running more tests for those boards.
+
+import sys
+import re, os
+from email.utils import parseaddr
+import sh
+import argparse
+from github import Github
+
+if "ZEPHYR_BASE" not in os.environ:
+    exit(1)
+
+repository_path = os.environ['ZEPHYR_BASE']
+sh_special_args = {
+    '_tty_out': False,
+    '_cwd': repository_path
+}
+
+def git(*args):
+    # Runs a git command using the 'sh' library (https://amoffat.github.io/sh/)
+
+    # Do not create a TTY for stdout, so that Git doesn't start a pager.
+    #
+    # Hack: Setting _cwd to the working dir seems pointless as of writing (it
+    # should be the default), but keep it around in case it's working around
+    # some issue
+    return sh.git(*args, _tty_out=False, _cwd=os.getcwd())
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+                description="Generate a sanitycheck argument for for boards "
+                            " that have changed")
+    parser.add_argument('-c', '--commits', default=None,
+            help="Commit range in the form: a..b")
+    parser.add_argument('-r', '--repo', default=None,
+                        help="Github repository")
+    parser.add_argument('-p', '--pull-request', default=0, type=int,
+                        help="Pull request number")
+
+    return parser.parse_args()
+
+def main():
+    boards = set()
+
+    args = parse_args()
+    if not args.commits:
+        exit(1)
+
+    commit = git("diff","--name-only", args.commits)
+    files = commit.split("\n")
+
+    docs = 0
+    images = 0
+    kernel = 0
+    code = 0
+    net = 0
+    bt = 0
+
+    matches = [
+            {
+                "area": "Modem",
+                "regex" : ["^drivers/modem"],
+                "counter": 0,
```

Comment:
just place the counters in a separate array of ints that is the same length as matches, then you have no need to add this to every single entry

Reply:
Ah but you'll need `OrderedDict` then. Up to you, you could also create a simple container class `Area` that has `name`, `regex`, `counter`, and `labels` as members

Reply:
that is script i wrote like a year ago very fast and just brought it back to life, I have no intention to keep this structure and maintain it here, idea is to move to the tree similar to CODEOWNERS later, so all of this will be scrapped.